### ISSUE TITLE
Discussion - AdvancedExpressionFolding for Kotlin

### ISSUE THREAD

**Issue by ciscorucinski** [cheptsov/AdvancedExpressionFolding#87]
Basically this same plugin for Kotlin. It would convert Kotlin syntax to Java Syntax as much as possible.

I understand a good amount of why Kotlin changed the syntax, and a lot of people say that if you can read Java code, then you can read Kotlin code. However, that is not always the case. This plugin would help those people read Kotlin code.

i.e. In Java, you put the Type first then the name of the object. This is reversed in Kotlin. This plugin would take the Kotlin variables and parameters and make the Type first and the name second. It would, of course, do a lot of other things, too.

Yes, the folds will be longer, but this is intended to help newcomers to the language. I would like to copy Kotlin code from Github readme files and have my IDE fold it into Java-like code (as a specific use-case). I think this new plugin would have an even larger impact on developers switching to Kotlin.

**Comment by aristotll** [comment]
I do not think it is necessary. This plugin is to reduce the syntax redundancy of Java.

**Comment by ciscorucinski** [comment]
@aristotll I agree about this plugin. This new idea would simply be a new plugin to help Kotlin Developers by folding things into its Java equivalent.

Right now I use this plugin and a secondary goal of this plugin appears to help developers transfer into the Kotlin language from Java.

This new plugin would help Kotlin developers coming from Java.
